### PR TITLE
show devs that they cannot grap the mac address of a device

### PR DESCRIPTION
**Pull request by isaidamier**

### CODE REVIEW

File: BluetoothLeChat/app/src/main/java/com/example/bluetoothlechat/scan/DeviceScanFragment.kt [hunk 1]
```diff
@@ -68,7 +69,8 @@ class DeviceScanFragment : Fragment() {
         savedInstanceState: Bundle?
     ): View? {
         _binding = FragmentDeviceScanBinding.inflate(inflater, container, false)
-
+        val devAddr = getString(R.string.your_device_address)+ ChatServer.getYourDeviceAddress()
```
Comment (codingjeremy) [inline BluetoothLeChat/app/src/main/java/com/example/bluetoothlechat/scan/DeviceScanFragment.kt:72]: Should be a space before + sign. Might want to run through Ktlint. It picks up all this stuff for you.

File: BluetoothLeChat/app/src/main/java/com/example/bluetoothlechat/scan/DeviceScanFragment.kt [hunk 2]
```diff
@@ -68,7 +69,8 @@ class DeviceScanFragment : Fragment() {
...
+        binding.yourDeviceAddr.text =devAddr
```
Comment (codingjeremy) [inline BluetoothLeChat/app/src/main/java/com/example/bluetoothlechat/scan/DeviceScanFragment.kt:73]: missing space after =

File: BluetoothLeChat/app/src/main/res/layout/fragment_device_scan.xml [hunk 3]
```diff
@@ -14,8 +14,7 @@
      See the License for the specific language governing permissions and
      limitations under the License.
 -->
-<FrameLayout
-    xmlns:android="http://schemas.android.com/apk/res/android"
+<FrameLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (codingjeremy) [inline BluetoothLeChat/app/src/main/res/layout/fragment_device_scan.xml:17]: If I put others all on a new line, I'm usually consistent. I would put this on a newline too to make it more readable.

File: BluetoothLeChat/app/src/main/res/layout/fragment_device_scan.xml [hunk 4]
```diff
@@ -24,37 +23,60 @@
...
             android:layout_width="wrap_content"
             android:layout_height="wrap_content"
-            android:layout_gravity="center"/>
+            android:layout_gravity="center" />
 
         <TextView
             android:layout_width="wrap_content"
             android:layout_height="wrap_content"
             android:layout_gravity="center"
+            android:paddingTop="16dp"
```
Comment (codingjeremy) [inline BluetoothLeChat/app/src/main/res/layout/fragment_device_scan.xml:39]: should these all be resources?

File: BluetoothLeChat/app/src/main/res/values/strings.xml [hunk 5]
```diff
@@ -31,4 +31,6 @@
     <string name="connect_device_prompt">Connect with nearby devices</string>
     <string name="send_message_button">Send Message</string>
     <string name="chatting_with_device">Chatting with %1$s</string>
+    <string name="your_device_address">Your device address:</string>
+    <string name="scanning_header">Choose a device to chat with:</string>
 </resources>
```
Comment (codingjeremy) [inline BluetoothLeChat/app/src/main/res/values/strings.xml:36]: Missing empty new line

File: BluetoothLeChat/app/src/main/res/layout/fragment_device_scan.xml [hunk 6]
```diff
@@ -24,37 +23,60 @@
...
         android:layout_height="match_parent"
-        android:visibility="gone"/>
+        android:orientation="vertical">
+
+        <TextView
+            android:id="@+id/your_device_addr"
+            style="@style/TextAppearance.AppCompat.Medium"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:padding="16dp"
```
Comment (codingjeremy) [inline BluetoothLeChat/app/src/main/res/layout/fragment_device_scan.xml:54]: dimens resource?

File: BluetoothLeChat/app/src/main/res/layout/fragment_device_scan.xml [hunk 7]
```diff
@@ -24,37 +23,60 @@
...
+            android:text="@string/your_device_address"
+            android:textStyle="bold" />
+
+        <TextView
+            style="@style/TextAppearance.AppCompat.Title"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:padding="16dp"
```
Comment (codingjeremy) [inline BluetoothLeChat/app/src/main/res/layout/fragment_device_scan.xml:62]: dimens?

File: BluetoothLeChat/app/src/main/res/layout/fragment_device_scan.xml [hunk 8]
```diff
@@ -24,37 +23,60 @@
...
 
         <TextView
             android:layout_width="wrap_content"
             android:layout_height="wrap_content"
             android:layout_gravity="center"
+            android:paddingTop="16dp"
             android:text="@string/active_scan_message"
-            android:textSize="18sp"
-            android:paddingTop="16dp"/>
+            android:textSize="18sp" />
```
Comment (codingjeremy) [inline BluetoothLeChat/app/src/main/res/layout/fragment_device_scan.xml:41]: dimens?

File: BluetoothLeChat/app/src/main/java/com/example/bluetoothlechat/scan/DeviceScanFragment.kt (hunk 1 above)
Comment (isaidamier) [reply BluetoothLeChat/app/src/main/java/com/example/bluetoothlechat/scan/DeviceScanFragment.kt:72]: ack

File: BluetoothLeChat/app/src/main/java/com/example/bluetoothlechat/scan/DeviceScanFragment.kt (hunk 2 above)
Comment (isaidamier) [reply BluetoothLeChat/app/src/main/java/com/example/bluetoothlechat/scan/DeviceScanFragment.kt:73]: ack

File: BluetoothLeChat/app/src/main/res/layout/fragment_device_scan.xml (hunk 4 above)
Comment (isaidamier) [reply BluetoothLeChat/app/src/main/res/layout/fragment_device_scan.xml:39]: yes. The project itself, which I inherited, has a number of those. I plan a follow on PR to fix them all.

File: BluetoothLeChat/app/src/main/res/values/strings.xml (hunk 5 above)
Comment (isaidamier) [reply BluetoothLeChat/app/src/main/res/values/strings.xml:36]: ack

File: BluetoothLeChat/app/src/main/res/layout/fragment_device_scan.xml (hunk 6 above)
Comment (isaidamier) [reply BluetoothLeChat/app/src/main/res/layout/fragment_device_scan.xml:54]: ditto. The project itself, which I inherited, has a number of those. I plan a follow on PR to fix them all.

File: BluetoothLeChat/app/src/main/res/layout/fragment_device_scan.xml (hunk 7 above)
Comment (isaidamier) [reply BluetoothLeChat/app/src/main/res/layout/fragment_device_scan.xml:62]: ditto

File: BluetoothLeChat/app/src/main/res/layout/fragment_device_scan.xml (hunk 8 above)
Comment (isaidamier) [reply BluetoothLeChat/app/src/main/res/layout/fragment_device_scan.xml:41]: ditto

File: BluetoothLeChat/app/src/main/res/layout/fragment_device_scan.xml (hunk 7 above)
Comment (codingjeremy) [reply BluetoothLeChat/app/src/main/res/layout/fragment_device_scan.xml:63]: ack

**Review by codingjeremy — APPROVED** [review]

**Review by fredchung — APPROVED** [review]

**Pull request merged**